### PR TITLE
Support integer types in TIR expression operators

### PR DESCRIPTION
**Pull request by jikechao**
This PR addresses the issue where `tvm.tir.exp` does not support integer types (e.g., int32, int64), causing an InternalError during LLVM code generation with the message.
The issue arises because the `llvm.exp` intrinsic expects floating-point inputs, but no type conversion is performed for integer inputs.

I opened this PR to solve it via type conversion. This change aligns the behavior of `tir.exp` with libraries like **PyTorch** and **NumPy**, which implicitly convert integer inputs to floating-point types for their exponential functions.

Fix https://github.com/apache/tvm/issues/18381

### CODE REVIEW

File: python/tvm/tir/op.py [hunk 1]
```diff
@@ -2116,6 +2116,8 @@ def exp(x):
         The result.
     """
     x = tir.convert(x)
+    if "int" in x.dtype:
+        x = tir.Cast("float64", x)
```
[the remark this answers is not in the document]
Comment (cbalint13) [reply python/tvm/tir/op.py]: Wouldn't be```float32``` just enough (I am not sure), it would be more HW friendly.

But in any case, please introduce a integer test to cover this ```tir.exp``` here:
https://github.com/apache/tvm/blob/main/tests/python/tir-base/test_tir_intrin.py#L67 

Lets make sure that slipping to integer domain will hold up with tests.

**Review by cbalint13 — CHANGES_REQUESTED** [review]

**Comment by jikechao** [comment]
> Idea for future,
> 
> Beside this casting to float (simplest) I think we could try using a LUT like implementation assuring that we truly emit integer ISA counter part from llvm (staying in the int domain), so we could also tackle on a good integer `softmax()` implementation, TVM being a compiler I would expect such capability.
> 
> See: https://github.com/google/gemmlowp/blob/16e8662c34917be0065110bfcd9cc27d30f52fdf/fixedpoint/fixedpoint.h#L781-L800



Hi @cbalint13, thanks for your review and suggestion. While LUT is well-suited for softmax due to its normalized output and constrained input range, implementing `exp(x)` for arbitrary integers is challenging and may not yield optimal performance because of the exponential function's vast output range and the complexity of fixed-point arithmetic for large values. 
For a quick fix, I prepared this patch as a workaround.

File: tests/python/tir-base/test_tir_intrin.py
Comment (cbalint13) [inline tests/python/tir-base/test_tir_intrin.py:121]: The ```exp``` is not tested with ints within whole ```test_unary_intrin()```.
* Add similar test like this exception here for the ```exp``` ints part, using ```np.randim.int()``` or similar
* Could also make an extra assertion that result is coming back as ```float32``` type always for now

**Review by cbalint13 — CHANGES_REQUESTED** [review]
